nntrainer/dataset/databuffer.h

```diff
@@ -133,6 +136,24 @@ class DataBuffer : public ml::train::Dataset {
    */
   const std::string getType() const;
 
+  /**
+   * @brief this function helps exporting the optimizer in a predefined format,
+   * while workarounding issue caused by templated function type eraser
+   *
+   * @param     exporter exporter that conatins exporting logic
+   * @param     method enum value to identify how it should be exported to
+   */
+  void exportTo(Exporter &exporter, const ExportMethods &method) const;
+
+  /**
+   * @brief check if given databuffer is exportable, this is needed because some
+   * data producer, mainly FuncDataProducer, cannot be serialized
+   *
+   * @param method proposed method
+   * @return bool true if serializable
+   */
+  bool isSerializable(const ExportMethods &method) const;
```

Comment:
where is this used?
what is the behavior when this results false - is there an empty dataset section or no dataset function?

Reply:
It will be no dataset section (in case of function callback), saving implementation missed using this function, now it's added